packages/components/src/DateTimePickers/InputDateTimePicker/date-extraction.test.js

```diff
@@ -43,44 +44,98 @@ describe('Date extraction', () => {
 		});
 	});
 
-	describe('extractPartsFromDateTime', () => {
-		it('should return empty parts on invalid date', () => {
+	describe('extractParts', () => {
+		it('should return empty parts on undefined value', () => {
 			// given
-			const invalidDate = 'lol';
 			const options = {
 				dateFormat: 'YYYY-MM-DD',
 				useTime: true,
 				useSeconds: true,
 			};
 
 			// when
-			const parts = extractPartsFromDateTime(invalidDate, options);
+			const parts = extractParts(undefined, options);
 
 			// then
 			expect(parts).toEqual({
 				date: undefined,
 				time: { hours: '', minutes: '', seconds: '' },
-				datetime: 'lol',
+				datetime: undefined,
 				textInput: '',
 			});
 		});
 
-		it('should return empty parts on undefined date', () => {
+		it('should return parts from timestamp', () => {
+			// given
+			const date = new Date(2015, 8, 15, 12, 58, 22);
+			const timestamp = date.getTime();
+			const options = { dateFormat: 'YYYY-MM-DD' };
+
+			// when
+			const parts = extractParts(timestamp, options);
+
+			// then
+			expect(parts).toEqual({
+				date: new Date(2015, 8, 15),
+				datetime: date,
+				textInput: '2015-09-15',
```

Comment:
8 or 9? there might be something that I don't get

Reply:
Same explanation